theano/tensor/basic.py

```diff
@@ -141,6 +141,8 @@ def as_tensor_variable(x, name=None, ndim=None):
         string.
     ndim : None or integer
         Return a Variable with this many dimensions.
+    dtype : numpy dtype or None
```

Comment:
Adding this parameter here will request that we respect it at all places in the method. I think it would be simpler to not add this parameter. You can call numpy.asarray(x, dtype=...) on the input before calling as_tensor_variable in this case.